Lib/fontTools/ttLib/tables/otTables.py

```diff
@@ -779,6 +779,17 @@ def preWrite(self, font):
 		format = 2
 		delta = None
 		for inID, outID in gidItems:
+			if inID is None or outID is None:
+				log.error(
+					"Substitution references one or more missing glyphs. "
+					"Check any items that contain a glyph ID of None:"
```

Comment:
why should the user check for Nones, can't the KeyError contain the name of the missing glyph?

Reply:
I thought it would be easier to find the feature that contains the error when a bit of context is given. It might be a mistake in the feature code, or a glyph that actually should be in the font, etc.

Reply:
"a mistake in the feature code" should be caught earlier than that

"a glyph that actually should be in the font" is a faulty font or a bug somewhere

Reply:
in general I don't like the combination of a logging ERROR message with a bare KeyError exception. There's no link between the two, and logging message can be ignored, the exception itself carries no message of its own